lib/ansible/modules/remote_management/manageiq/manageiq_group.py

```diff
@@ -0,0 +1,601 @@
+            else:
+                new_filters_resource['belongsto'] = list(new_belongsto_set)
+            filters_updated = True
+
+        # Process belongsto managed filter tags
+        if 'managed' in current_filters:
+            current_managed = current_filters['managed']
+        else:
+            current_managed = None
+
+        if self.have_same_values_nested_array(current_managed, managed_filters):
+            new_filters_resource['managed'] = current_filters['managed']
+        else:
+            if managed_filters_merge_mode == 'merge':
+                new_filters_resource['managed'] = self.merge_values_nested_array(current_managed, managed_filters)
+            else:
+                new_filters_resource['managed'] = managed_filters
+            filters_updated = True
+
+        if not filters_updated:
+            return None
+
+        return new_filters_resource
+
+    def create_group(self, description, role, tenant, managed_filters, belongsto_filters):
+        """ Creates the group in manageiq.
+
+        Returns:
+            the created group id, name, created_on timestamp,
+            updated_on timestamp.
+        """
+        # check for required arguments
+        for key, value in dict(description=description).items():
+            if value in (None, ''):
+                self.module.fail_json(msg="missing required argument: %s" % key)
+
+        url = '%s/groups' % self.api_url
+
+        resource = {'description': description}
+
+        if role is not None:
+            resource['role'] = dict(id=role['id'])
+
+        if tenant is not None:
+            resource['tenant'] = dict(id=tenant['id'])
+
+        if managed_filters or belongsto_filters:
+            resource['filters'] = {'managed': managed_filters, "belongsto": belongsto_filters}
+
+        try:
+            result = self.client.post(url, action='create', resource=resource)
+        except Exception as e:
+            self.module.fail_json(msg="failed to create group %s: %s" % (description, str(e)))
+
+        return dict(
+            changed=True,
+            msg="successfully created group %s" % description,
+            group_id=result['results'][0]['id']
+        )
+
+    @staticmethod
+    def create_result_group(group):
+        """ Creates the ansible result object from a manageiq group entity
+
+        Returns:
+            a dict with the group id, description, role_name, tenant_name, filters, group_type, created_on, updated_on
+        """
+        try:
+            role_name = group['miq_user_role_name']
+        except AttributeError:
+            role_name = None
+
+        if 'filters' in group['entitlement']:
+            filters = group['entitlement']['filters']
+        else:
+            filters = None
+
+        return dict(
+            id=group['id'],
+            description=group['description'],
+            role_name=role_name,
+            tenant_name=group['tenant']['name'],
+            filters=filters,
```

Comment:
I think it's nice for ansible returned structure to be similar to module arguments where possible.
Can you expand this into `managed_filters` and `belongsto_filters`?

Reply:
Yes, I will change this.